lib/capistrano/gitflow/helpers/helper.rb

```diff
@@ -216,5 +216,23 @@ def gitflow_tag_production
       set :branch, new_production_tag
     end
 
+    def gitflow_cleanup_tags
+      return if fetch(:gitflow_keep_tags).nil?
+      tags = `git for-each-ref --format='%(*committerdate:raw)%(committerdate:raw) %(refname)' refs/tags |   sort -n  | awk '{print $3}'`
+      tags = tags.split
+      if tags.count >= fetch(:gitflow_keep_tags)
+        puts "Keeping #{fetch(:gitflow_keep_tags)} Tags from total #{tags.count}"
+        tags_to_delete = (tags - tags.last(fetch(:gitflow_keep_tags)))
+        if tags_to_delete.any?
+          tags_without_refs = tags_to_delete.map{|tag| tag.gsub('refs/tags/', '') }
+          system "echo #{tags_without_refs.join(' ')} | xargs git tag -d"
```

Comment:
"git tag -d" accepts only one tag as parameter, that's why i used "xargs" to simplify a bit the code, so that i don't have to use a loop statement,